### DISCUSSION TITLE
Blog Next.js + Strapi HCMS

### DISCUSSION THREAD

**Discussion by AlexTechNoir** [strapi/strapi#7865] [Legacy Topics]
Hello, everyone!

I made a static demo blog with Next.js and Strapi Headless CMS. In the repo's readme you can find list of dependencies and detailed description of how I made some features. Study it if you find this interesting.

Demo: https://brave-northcutt-60bde9.netlify.app/

Repo: https://github.com/AlexTechNoir/Next.js-Strapi-Blog

**Comment by derrickmehaffy** [comment]
Thank you for sharing!

@malgamves

**Reply by AlexTechNoir** [reply]
You're welcome!

**Comment by ChristopherLMiller** [comment]
Very nice!  Gonna take a look further at your code because you use a lot of things I do and sometimes Im not sure if I'm going about things in a good manner.  Compare and see!

**Reply by AlexTechNoir** [reply]
Great! Enjoy!

**Reply by ChristopherLMiller** [reply]
I actually took inspiriation from your ProgressiveImage component, used it as a basis and modified to be more suitable for my specific need if you don't mind.  I've been after basically that just didn't want to spend the time trying to figure out how to implement so thats a great help.

**Reply by AlexTechNoir** [reply]
I don't mind, of course. I myself took this from some old class-based example and adapted to hooks. The only thing I couldn't figure out though is how to display a small image with the same big size as a big image initially, so the text couldn't jump down suddenly (in case when page load speed is extremely slow). It's kinda tricky if we need images to be responsive. Also, when I was building this demo I didn't know that styled-components can manipulate attributes in elements. This can be a game changer.

**Comment by malgamves** [comment]
This looks great! Wanna add it to our community [showcase](https://github.com/strapi/community-content/tree/master/showcase)?

**Reply by AlexTechNoir** [reply]
Thanks, Daniel! I see the real websites in this list. If demos are allowed, I will.

**Comment by malgamves** [comment]
You are very right @AlexTechNoir - you can make it a starter and submit it to the [starter section](https://github.com/strapi/community-content/tree/master/starters) of our community content repo. That way other people can use it too.

**Reply by AlexTechNoir** [reply]
Will do, thanks!
